src/components/MAPI/clusters/CreateClusterAppBundles/schemaUtils.ts

```diff
@@ -53,24 +53,72 @@ const formPropsProviderCAPA: Record<string, FormPropsPartial> = {
   0: {
     uiSchema: {
       'ui:order': [
-        'clusterName',
-        'clusterDescription',
-        'organization',
+        'metadata',
+        'providerSpecific',
         'controlPlane',
+        'connectivity',
+        'nodePools',
         '*',
       ],
-      bastion: {
-        instanceType: {
-          'ui:widget': InstanceTypeWidget,
-        },
+      baseDomain: {
+        'ui:widget': 'hidden',
       },
-      clusterName: {
-        'ui:widget': ClusterNameWidget,
+      connectivity: {
+        'ui:order': ['sshSsoPublicKey', '*'],
+        bastion: {
+          instanceType: {
+            'ui:widget': InstanceTypeWidget,
+          },
+        },
       },
       controlPlane: {
+        'ui:order': [
+          'apiMode',
+          'instanceType',
+          'replicas',
+          'rootVolumeSizeGB',
+          'etcdVolumeSizeGB',
+          'kubeletVolumeSizeGB',
+          'containerdVolumeSizeGB',
+          'oidc',
+          '*',
+        ],
         instanceType: {
           'ui:widget': InstanceTypeWidget,
         },
+        oidc: {
+          'ui:order': ['issuerUrl', 'clientId', '*'],
+        },
+      },
+      'cluster-shared': {
+        'ui:widget': 'hidden',
+      },
+      defaultMachinePools: {
+        'ui:widget': 'hidden',
+      },
+      kubectlImage: {
+        'ui:widget': 'hidden',
+      },
+      managementCluster: {
+        'ui:widget': 'hidden',
+      },
+      metadata: {
+        'ui:order': ['name', 'description', '*'],
+        name: {
+          // TODO: fix crash
+          //'ui:widget': ClusterNameWidget,
```

Comment:
@gusevda Can you please help me with this one? If I uncomment line 109, the app crashes ("can't read property length of undefined", if I remember correctly).

Reply:
you should also change `formData` property below to match the new structure. It should be
```
formData: (clusterName, organization) => {
    return {
        metadata: {  
            name: clusterName,
            organization,
        }
    };
},
```

Reply:
Thanks, that worked.